src/lib/operators/table_scan/base_single_column_table_scan_impl.cpp

```diff
@@ -39,24 +39,50 @@ void BaseSingleColumnTableScanImpl::handle_segment(const ReferenceSegment& segme
   const ChunkID chunk_id = context->_chunk_id;
   auto& matches_out = context->_matches_out;
 
+  const auto& pos_list = segment.pos_list();
+
+  if (pos_list->references_single_chunk() && !pos_list->empty()) {
+    // Fast path :)
+
+    const auto chunk = segment.referenced_table()->get_chunk((*pos_list)[0].chunk_id);
```

Comment:
> Do you think it is worth wrapping (*pos_list)[0].chunk_id into PosList::get_single_referenced_chunk()

Mmh, maybe `std::optional<ChunkID> PosList::common_chunk_id` and get rid of `references_single_chunk`?

> And, one day we could stop storing the ChunkID in for every entry in a single-chunk PosList

I don't think that will work for the reason named below. If we introduce variation here, we would have two different types of PosList and might have to convert between the two. That conversion is likely more expensive than wasting a couple of ChunkIDs.

Reply:
Decided against the `optional` because there might be two cases in which we don't have common chunk ID: (1) that property has not been guaranteed and (2) the pos list is empty.

Simply added `ChunkID common_chunk_id` which fails if those two conditions have not been excluded by the caller.